sdk/@launchdarkly/observability-python/ldobserve/__init__.py

```diff
@@ -1,7 +1,132 @@
-from typing import Optional
-
+import logging
+import os
+from typing import List, Optional
+from ldclient.hook import Hook
+from opentelemetry.instrumentation import auto_instrumentation
 from ldobserve.config import ObservabilityConfig
+from ldobserve.observe import _ObserveInstance
+import ldobserve.observe as observe
+from ldobserve.config import _ProcessedConfig
+import ldobserve.observe
+from ldobserve._otel.configuration import _OTELConfiguration
+from ldclient.plugin import Plugin, EnvironmentMetadata, PluginMetadata
+from ldotel.tracing import Hook, HookOptions
+from ldclient.client import LDClient
+from opentelemetry.instrumentation.environment_variables import (
+    OTEL_PYTHON_DISABLED_INSTRUMENTATIONS,
+)
+from opentelemetry.sdk.environment_variables import OTEL_EXPERIMENTAL_RESOURCE_DETECTORS
+
+
+def _extend_environment_list(env_var_name: str, *new_items: str) -> None:
+    """
+    Extend an environment variable containing a comma-separated list with additional items.
+
+    This function reads the current value of the specified environment variable,
+    adds the provided items to the comma-separated list, and sets the
+    environment variable with the extended list.
+
+    Args:
+        env_var_name: The name of the environment variable to extend.
+        *new_items: Variable number of items to add to the list.
+
+    Example:
+        >>> _extend_environment_list("MY_LIST", "item1", "item2")
+        # If MY_LIST was "existing_item", it will become "existing_item,item1,item2"
+    """
+    current_value = os.getenv(env_var_name, "")
+
+    # Split the current value by comma and strip whitespace
+    current_list = [item.strip() for item in current_value.split(",") if item.strip()]
+
+    # Add new items, avoiding duplicates
+    for item in new_items:
+        if item.strip() and item.strip() not in current_list:
+            current_list.append(item.strip())
```

Comment:
Couple of performance questions / comments.

**Local variable**

Do we think this is worth it?

```suggestion
        cleaned_item = item.strip()
        if cleaned_item and cleaned_item not in current_list:
            current_list.append(cleaned_item)
```

**List vs Set**

We are doing a `not in` against a list. If we built-up `current_list` as a set, this check would be faster (theoretically). But if we think the lists are going to be super short, this likely won't matter much in practice.